src/concepts/plugins/frontend.md

```diff
@@ -0,0 +1,479 @@
+- `types.ts`: This file defines and exports type delcarations/definitions to be bundled with your code and be consumable by dependents.
+
+::: info
+You may only have one CSS and entrypoint file per plugin package.
+:::
+
+### manifest.json
+
+The example manifest file in the frontend starterkit is as follows:
+
+::: tip Key/Value Breakdown
+
+```json
+{
+  "id": "starterkit-plugin",
+  "name": "StarterKit Plugin",
+  "version": "0.1.0",
+  "description": "This is a starterkit plugin",
+  "author": {
+    "name": "Caido Labs Inc.",
+    "email": "dev@caido.io",
+    "url": "https://github.com/caido/starterkit-plugin"
+  },
+  "plugins": [
+    {
+      "kind": "frontend",
+      "id": "caido-extra-commands",
+      "name": "Extra Commands",
+      "entrypoint": "frontend/script.js",
+      "style": "frontend/style.css"
+    }
+  ]
+}
+
+```
+
+Within the first data object:
+
+- **REQUIRED**: `id` - Must be **unique** and must only consist of **lowercase** letters, **numbers**, **hyphens** and **underscores** (_the order of which must satisfy the regex: `^[a-z]+(?:[_-][a-z0-9]+)*$`).
+- **OPTIONAL**: `name` - If not supplied, the `id` will be used as the `name`. This property is not subject to the same rules of the `id` property.
+- **REQUIRED**: `version` - Versioning follows the `MAJOR.MINOR.PATCH` syntax.
+- **OPTIONAL**: `description`
+- **OPTIONAL**: `author`
+
+Within the `plugins` array:
+
+- **REQUIRED**: `kind` - This property specifies the Plugin type: `frontend`, `backend` or `fullstack`.
+- **REQUIRED**: `id` - Must be **unique** and must only consist of **lowercase** letters, **numbers**, **hyphens** and **underscores** (_the order of which must satisfy the regex: `^[a-z]+(?:[_-][a-z0-9]+)*$`).
+- **OPTIONAL**: `name` - If not supplied, the `id` will be used as the `name`. This property is not subject to the same rules of the `id` property.
+- **REQUIRED**: `entrypoint` - This property specifies the location of the primary script to be executed when the Caido application/Plugin is launched.
+- **OPTIONAL**: `style` - This property specifies the location of the CSS file to be used to stylize elements of your Plugin.
+
+Additional information:
+
+- Multiple plugins are allowed in the array of each `manifest.json` file.
+- The first instance of the `name` property is the cosmetic Plugin package name and is displayed in the [Plugins](/reference/features/workspace/plugins.md) table. If the array includes multiple Plugins - additional instances of the `name` property are displayed next to the Plugins when expanding the parent element by toggling the `>` icon to `∨`.
+- The `frontend` directory is generated upon the completion of the build process when using the Vite build tool.
+:::
+
+::: info
+
+The specification for the `manifest.json` file is set in the [schema.json](https://github.com/caido/plugin-manifest/blob/main/src/schema.json) file.
+
+Expand the sections below to view the `schema.json` file:
```

Comment:
I would prefer we dont put the spec here since it's another place we need to update.

Reply:
We should explain to users how to use the plugin-manifest package instead. Though not now (leave a note in the doc).